spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonLoadBalancedRetryPolicy.java

```diff
@@ -105,17 +118,14 @@ public void registerThrowable(LoadBalancedRetryContext context, Throwable throwa
         //if this is a circuit tripping exception then notify the loadbalancer
     	if (lbContext.getRetryHandler().isCircuitTrippingException(throwable)) {
     		ServiceInstance serviceInstance = context.getServiceInstance();
-    		for (Server lbServer : lbContext.getLoadBalancer().getAllServers()) {
-    			if (lbServer.getHost().equals(serviceInstance.getHost())
-    					&& lbServer.getPort() == serviceInstance.getPort()){
-    				ServerStats serverStats = lbContext.getServerStats(lbServer);
-    				serverStats.incrementSuccessiveConnectionFailureCount();
-    				serverStats.addToFailureCount();    				
-    				logger.debug(lbServer.getHostPort() + " RetryCount: " + context.getRetryCount() 
-    					+ " Successive Failures: " + serverStats.getSuccessiveConnectionFailureCount() 
-    					+ " CirtuitBreakerTripped:" + serverStats.isCircuitBreakerTripped());
-    				break;
-    			}
+    		Server lbServer = ribbonServerListChangeListener.getServer(serviceInstance.getHost() + ":" + serviceInstance.getPort());
```

Comment:
I would like to see this code encapsulated in its own method

Reply:
thanks.  I will make that change